File: frontend/src/app/infrastructure/api/models/release-detail.model.ts
import { ReleaseBandModel } from './release-band.model';
import { ReleaseTrackModel } from './release-track.model';

export interface ReleaseDetailModel {
  id: string;
  title: string;
  releaseType: string;
  releaseDate: string | null;
  year: number | null;
  labelText: string | null;
  coverImageUrl: string | null;
  band: ReleaseBandModel | null;
  formats: string[];
  tracks: ReleaseTrackModel[];
}

File: backend/src/DiyMusicCommunity.Application/Releases/GetReleaseDetail/ReleaseDetailModel.cs
using DiyMusicCommunity.Domain.Enums;

namespace DiyMusicCommunity.Application.Releases.GetReleaseDetail;

public sealed class ReleaseDetailModel
{
    public Guid Id { get; init; }
    public string Title { get; init; } = string.Empty;
    public ReleaseType ReleaseType { get; init; }
    public DateOnly? ReleaseDate { get; init; }
    public int? Year { get; init; }
    public string? LabelText { get; init; }
    public string? CoverImageUrl { get; init; }
    public ReleaseBandModel? Band { get; init; }
    public IReadOnlyList<Format> Formats { get; init; } = [];
    public IReadOnlyList<ReleaseTrackModel> Tracks { get; init; } = [];
}
